src/bin/ion/commands/beta/generate/model.rs

```diff
@@ -0,0 +1,268 @@
+/// and sequence type is described by nested `sequence_type` (e.g. List or SExp).
+/// If there is no `element` constraint present in schema type then `element_type` will be None.
+/// If there is no `type` constraint present in schema type then `sequence_type` will be None.
+/// e.g. Given below ISL,
+/// ```
+/// type::{
+///   name: sequence_type,
+///   element: int
+/// }
+/// ```
+/// Corresponding generated code in Rust would look like following:
+/// ```
+/// struct SequenceType {
+///    value: Vec<i64>
+/// }
+/// ```
+#[allow(dead_code)]
+#[derive(Debug, Clone, PartialEq, Serialize)]
+pub struct Sequence {
+    // Represents the fully qualified name for this data model
+    name: FullyQualifiedTypeName,
+    // Represents doc comment for the generated code
+    doc_comment: String,
+    // Represents the fully qualified name with namespace where each element of vector stores a module name or class/struct name.
+    // _Note: that a hashmap with (FullQualifiedTypeReference, DataModel) pairs will be stored in code generator to get information on the element_type name used here._
+    element_type: FullyQualifiedTypeReference,
+    // Represents the type of the sequence which is either `sexp` or `list`.
+    sequence_type: SequenceType,
+    // Represents the source ISL type which can be used to get other constraints useful for this type.
+    // For example, getting the length of this sequence from `container_length` constraint or getting a `regex` value for string type.
+    // This will also be useful for `text` type to verify if this is a `string` or `symbol`.
+    // TODO: `IslType` does not implement `Serialize`, define a custom implementation or define methods on this field that returns values which could be serialized.
+    #[serde(skip_serializing)]
+    source: IslType,
+}
+
+/// Represents a collection of field name/value pairs (e.g. a map)
+/// e.g. Given below ISL,
+/// ```
+/// type::{
+///   name: struct_type,
+///   fields: {
+///      a: int,
+///      b: string,
+///   }
+/// }
+/// ```
+/// Corresponding generated code in Rust would look like following:
+/// ```
+/// struct StructType {
+///    a: i64,
+///    b: String,
+/// }
+/// ```
+#[allow(dead_code)]
+#[derive(Debug, Clone, PartialEq, Serialize)]
+pub struct Structure {
+    // Represents the fully qualified name for this data model
+    name: FullyQualifiedTypeName,
+    // Represents doc comment for the generated code
+    doc_comment: String,
+    // Represents whether the struct has closed fields or not
+    is_closed: bool,
+    // Represents the fields of the struct i.e. (field_name, field_value) pairs
+    // field_value represents the type of the value field as fully qualified name
+    // _Note: that a hashmap with (FullQualifiedTypeReference, DataModel) pairs will be stored in code generator to get information on the field_value name used here._
+    fields: HashMap<String, FullyQualifiedTypeReference>,
```

Comment:
We should have a TODO to add something like this so that we can distinguish optional and required fields.
```rust
enum FieldRequiredness {
    Required,
    Optional,
}

struct FieldReference(FullyQualifiedTypeReference, FieldRequiredness)
```